dcos_test_utils/dcos_api_session.py

```diff
@@ -323,12 +323,28 @@ def _wait_for_srouter_slaves_endpoints(self):
                     retry_on_result=lambda r: r is False,
                     retry_on_exception=lambda _: False)
     def _wait_for_metronome(self):
+        # Although this is named `wait_for_metronome`, some of the waiting
+        # done in this function is, implicitly, for Admin Router.
         r = self.get('/service/metronome/v1/jobs')
-        # 500 and 504 are the expected behavior of a service
-        # backend that is not up and running.
-        if r.status_code == 500 or r.status_code == 504:
-            log.info("Metronome gateway timeout, continue waiting for backend...")
+        expected_error_codes = {
+            404: ('It may be the case that Admin Router is returning a 404 '
+                  'despite the Metronome service existing because it uses a cache. '
+                  'This cache is updated periodically.'),
+            503: ('The cache is not working. This means that either Marathon '
```

Comment:
I am not really convinced about the absolute correctness of this error message in all cases where Admin Router emits a 503. In such cases I _strongly_ favor having neutral error messages that have a low risk of providing false/misleading conclusions.